### PR TITLE
Better docker ☁📦

### LINKED ISSUE #16: Leverage docker-xvfb
It would be great to take the learnings from [docker-xvfb](https://github.com/bengreenier/docker-xvfb) and apply them here. As a part of that project, we've been working to get a cleaner electron run, based on X itself.

If we can even be based on those docker images without a regression in [this x version/fix](https://github.com/bengreenier/browserd/pull/9) that would be even better imo.

### LINKED ISSUE #25: Support electron-builder 🏗
We have partial configuration for electron-builder, we should fully implement support for it.

### PR DESCRIPTION
**Pull request by bengreenier**
Improves our docker setup to rely on fewer custom-built dependencies (and instead to use other, existing projects/learnings).

+ Close #16 
+ Fix #25

This also makes electron much happier at boot 😸

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,15 +12,37 @@
     "pack": "npm run build && electron-builder --dir",
     "dist": "npm run build && electron-builder"
   },
-  "author": "",
-  "license": "ISC",
+  "author": {
```
Comment (bengreenier) [inline package.json]: @phongcao @conrad465 if you'd like to be in this least, please feel free to add a commit to this branch with your info! 🚀

File: src/__tests__/browser/main.spec.ts [hunk 2]
```diff
@@ -1,10 +1,14 @@
+import pino from "pino";
 import {
-  K_BROWSER_STORAGE,
+  K_BROWSER_STORAGE, K_PRELOAD_LOGGER_KEY,
```
Comment (bengreenier) [inline src/__tests__/browser/main.spec.ts]: nit: alignment

**Comment by bengreenier** [comment]
actually @phongcao @conrad465 let's hold off reviewing this - manual testing is yielding more strangeness than i anticipated.

**Comment by bengreenier** [comment]
Manual testing completed - this gets us feature parity with what we have, but with slimmer deps and a happier electron process.

Still would love (seriously) to spend time looking at #30 - but probably outside of the scope for now.

File: package.json (hunk 1 above)
Comment (phongcao) [reply package.json]: Added mine. @conrad465 Please add yourself to the list.

**Comment by phongcao** [comment]
@bengreenier Any idea why it looks being scaled when running in Docker?
bug

File: Dockerfile [hunk 3]
```diff
@@ -1,45 +1,41 @@
...
+# we start as root, but quickly drop to "electron"
+USER root
+# as root:
+# start dbus system
+# as electron:
+# start dbus session
+# start x11 session (xvfb)
+# start browserd
+CMD service dbus start \
+  && su electron -c 'dbus-run-session -- xvfb-run -s "-ac -br -nocursor -screen 0 1920x1080x24 -nolisten tcp" browserd --no-sandbox --disable-gpu'
```
Comment (phongcao) [inline Dockerfile]: Ah, so we've changed the resolution to 1920x1080.

Comment (phongcao) [reply Dockerfile]: Should we read these values from `.env`?

Comment (bengreenier) [reply Dockerfile]: no. this is build-time config, not run-time config. its up to the user to manually configure this if they want other values.

Comment (phongcao) [reply Dockerfile]: Then I'd love this resolution settings is mentioned somewhere in our docs.

File: Dockerfile [hunk 4]
```diff
@@ -1,45 +1,41 @@
-# build, lint, and test
+# lint, test, transpile and buid binary (as deb)
```
Comment (phongcao) [inline Dockerfile]: typo: build

**Comment by bengreenier** [comment]
@phongcao thanks, but there's no need for you to block due to conflicts, github/git will block that on it's own 😸 - if this meets your bar to ship, please approve.

**Pull request merged**